### PR TITLE
Allow Swift to enable LLDB tests in Windows CI

### PR DESCRIPTION
**Pull request by weliveindetail**
This is the companion PR with all the llvm-project changes for https://github.com/swiftlang/swift/pull/76894

### CODE REVIEW

**Comment by hjyamauchi** [comment]
@swift-ci please test

**Comment by weliveindetail** [comment]
Build failed because make wasn't found. We don't have detection for it until the swift PR lands and I missed the cherry-pick that turns it into a warning.

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by hjyamauchi** [comment]
@swift-ci please test Windows Platform

**Comment by weliveindetail** [comment]
Windows platform failed due to an unrelated build error. LLDB was building fine. We didn't get to the test stage.

**Comment by compnerd** [comment]
@swift-ci please test Windows platform

File: lldb/bindings/python/CMakeLists.txt [hunk 1]
```diff
@@ -198,6 +198,24 @@ function(finish_swig_python swig_target lldb_python_bindings_dir lldb_python_tar
...
+          "Your CMake version is ${CMAKE_VERSION}. In order to run LLDB tests "
+          "on Windows please upgrade to 3.21.0 at least (or disable tests with "
+          "LLDB_INCLUDE_TESTS=Off)")
+      endif()
+    else()
+      add_custom_command(TARGET ${swig_target} POST_BUILD
+        COMMAND ${CMAKE_COMMAND} -E copy -t ${lldb_python_target_dir} $<TARGET_RUNTIME_DLLS:liblldb>
+        COMMAND_EXPAND_LISTS)
+    endif()
+  endif()
```
Comment (JDevlieghere) [inline lldb/bindings/python/CMakeLists.txt:218]: Any reason this can't go upstream? Or is this a cherrypick?

Comment (weliveindetail) [reply lldb/bindings/python/CMakeLists.txt:218]: In upstream LLVM liblldb has no dependent DLLs (except Python itself) so there was no case for it. And yes, it's a cherry-pick from https://github.com/swiftlang/llvm-project/pull/9370

**Review by JDevlieghere — APPROVED** [review]
LGTM. Most of these are cherrypicks from upstream and everything else is Swift specific. Thanks @weliveindetail!

**Pull request merged**